pkg/segment/query/segquery.go

```diff
@@ -831,8 +831,9 @@ func applyAggOpOnSegments(sortedQSRSlice []*QuerySegmentRequest, allSegFileResul
 		aggHasEvalFunc := segReq.aggs.HasValueColRequest()
 		aggHasValuesFunc := segReq.aggs.HasValuesFunc()
 		aggHasListFunc := segReq.aggs.HasListFunc()
+		aggHasMinMaxFunc := segReq.aggs.HasMinMaxFunc() // TODO: Add support for min/max strings in sst
 		var sstMap map[string]*structs.SegStats
-		if searchType == structs.MatchAllQuery && isSegmentFullyEnclosed && !aggHasEvalFunc && !aggHasValuesFunc && !aggHasListFunc {
+		if searchType == structs.MatchAllQuery && isSegmentFullyEnclosed && !aggHasEvalFunc && !aggHasValuesFunc && !aggHasListFunc && !aggHasMinMaxFunc {
```

Comment:
nit: this condition is getting long. Maybe extract it into a function (so the function makes the purpose clear) or add a comment saying what this condition means

Reply:
addressed in the latest commit.